### ISSUE TITLE
`sudo gem install fastlane` fails installing jwt on macOS 10.12

### ISSUE THREAD

**Issue by kballard** [fastlane/fastlane#11411]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
The fastlane installation says `sudo gem install fastlane` works on macOS. However, fastlane depends on `jwt`, and `jwt` version 2.0 or higher requires Ruby 2.1 or higher. Unfortunately, macOS 10.12 only ships Ruby 2.0, so fastlane fails to install there. I can work around this issue by running `sudo gem install -v 1.5.6 jwt` first, but this is not immediately obvious.

##### Complete output when running fastlane, including the stack trace and command used
N/A

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.12.6                                                        |
| Ruby                        | 2.0.0                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.14.3 (Apple Git-98)                              |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.12.6 (16G1114)                                     |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib |
| OpenSSL Version             | OpenSSL 0.9.8zc 19 Mar 2015                                    |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 9.2                                                            |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.72.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.2        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.4.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.7.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.60.0       |
| faraday                   | 0.13.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.1        |
| gh_inspector              | 1.0.3        |
| json                      | 1.7.7        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.2       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.2.1        |
| bundler                   | 1.16.1       |
| faraday_middleware        | 0.12.2       |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.6       |
| libxml-ruby               | 2.6.0        |
</details>

*generated on:* **2018-01-03**
</details>

**Comment by tatemz** [comment]
Same here. Is there any version of Fastlane that is known to work with Ruby 2.0?

**Comment by trusk89** [comment]
Same.

**Comment by KrauseFx** [comment]
The awesome @jdee actually worked on a PR for this one https://github.com/fastlane/fastlane/pull/11374

Can you confirm that you're on Sierra, and not on High Sierra yet?

**Comment by trusk89** [comment]
I can confirm my CI machine that has this issue is on Sierra.

**Comment by KrauseFx** [comment]
Looks like the only solution for now is

```
[sudo] gem install jwt -v 1.5.6
```

or define the `1.5.6` dependency if you use a `Gemfile`

**Comment by jdee** [comment]
You don't need to do anything special with a Gemfile. The Bundler automatically accounts for the Ruby version and will install the right jwt version. You don't need to explicitly declare jwt at all in your Gemfile. Just

```Ruby
gem "fastlane"
```

And then `bundle install` or `bundle update`.

The problem is that RubyGems and the Bundler are two halves to the solution. The `gem install` command is basically a lower-level utility that may require you to manually manage dependencies like this. The Bundler does that for you.

It's hard to add a pin to the gemspec. If we do, then `gem install` will succeed with Ruby 2.0, but it will fail for anyone who already has jwt 2.x installed, which is possibly most users of Fastlane. Those users would have to manually remove all 2.x versions of jwt first.

Ruby 2.0 has been deprecated for nearly two years, but it was shipped with Sierra anyway.

In short, the solutions are:

1. Upgrade to High Sierra, which ships Ruby 2.3.3.
2. Use a Ruby version manager (rbenv/RVM/chruby) to install a more current version of Ruby and not use the system Ruby on Sierra.
3. Use a Gemfile and the Bundler.
4. `[sudo] gem install jwt -v 1.5.6`

Maybe that needs to get into the docs somewhere.

**Comment by trusk89** [comment]
@jdee I'm doing a gemfile with fastlane and it doesn't install the correct version.
